src/utils/ucx/ucx_utils.cpp

```diff
@@ -406,15 +406,15 @@ nixlUcxWorker::nixlUcxWorker(std::shared_ptr<nixlUcxContext> &_ctx): ctx(_ctx)
         worker_params.thread_mode = UCS_THREAD_MODE_MULTI;
         break;
     default:
-        assert(ctx->mt_type < NIXL_UCX_MT_MAX);
-        abort();
+        NIXL_FATAL << "Invalid UCX worker type: " << ctx->mt_type;
     }
 
     status = ucp_worker_create(ctx->ctx, &worker_params, &worker);
-    if (status != UCS_OK)
-    {
-       // TODO: MSW_NET_ERROR(priv->net, "failed to create ucp_worker (%s)\n", ucs_status_string(status));
-        return;
+    if (status != UCS_OK) {
+        auto err_str = std::string("Failed to create UCX worker: ") +
+                       ucs_status_string(status);
+        NIXL_ERROR << err_str;
```

Comment:
IMO this PR should just add error logs and we need to introduce exception error handling as broader scope change

Reply:
https://github.com/ai-dynamo/nixl/blob/498fe2ac0b36f7223c8d9c66508de814928602d3/src/utils/ucx/ucx_utils.cpp#L163

Reply:
ok